scripts/aes_gcm_test.py

```diff
@@ -0,0 +1,77 @@
+# importing the module
+import json
+
+modes = {'encrypt': 0, 'decrypt': 1}
+
+def to_compound_str(name, val):
+    assert len(val) % 2 == 0, "Only even sized values supported"
+    if len(val) > 0:
+        import re
+        a = re.findall('..', val)
+        b = name + " = (const uint8_t []){ "
+        for s in a:
+            b += "0x" + s + ", "
+        b += "},\n\t" + name + "_len = " + repr((int)(len(val) / 2)) + ","
+    else:
+        b = name + " = NULL,\n\t" + name + "_len = 0,"
+    return b
+
+
+def generate_case(outf, tv, mode, line_num):
+    outf.write('{\n\t.algo = TEE_ALG_AES_GCM, .mode = ' + mode + ', .key_type = TEE_TYPE_AES,\n')
+    outf.write('\t' + to_compound_str('.key', tv['key']) + '\n')
+    outf.write('\t' + to_compound_str('.nonce', tv['iv']) + '\n')
+    outf.write('\t.aad_incr = 0,\n')
+    outf.write('\t' + to_compound_str('.aad', tv['aad']) + '\n')
+    outf.write('\t.in_incr = 0,\n')
+    outf.write('\t' + to_compound_str('.ptx', tv['msg']) + '\n')
+    outf.write('\t' + to_compound_str('.ctx', tv['ct']) + '\n')
+    outf.write('\t' + to_compound_str('.tag', tv['tag']) + '\n')
+    outf.write('\t.line = ' + repr(line_num) + ',\n')
```

Comment:
I suppose `'\t.line = __LINE__\n'` would work equally well (and perhaps be even more trustworthy :wink:) and would avoid the extra complexity of dealing with the `line_num` argument.

Reply:
smarter indeed :+1: